universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java

```diff
@@ -191,6 +192,43 @@ public boolean hasEnumerator(long enumeratorId) {
         .collect(ImmutableList.toImmutableList());
   }
 
+  /**
+   * Returns a list of the question definitions that may be used to define predicates on the block
+   * definition with the given ID.
+   *
+   * <p>The available question definitions for predicates satisfy BOTH of the following:
+   *
+   * <ul>
+   *   <li>In a block definition that comes sequentially before the block definition with the given
+   *       ID.
+   *   <li>In a block definition that has the same enumerator ID as the block definition with the
+   *       given ID.
```

Comment:
Issue (blocking): a repeated block should be able to specify a non-repeated (top level) question with its predicate.

Reply:
Ah, will do.

I thought we were doing an initial implementation without this. The complexity won't show up here, but in the predicate evaluation, I believe.

Reply:
Done!